### PR TITLE
Nomad py integration tests dockerized

### PR DESCRIPTION
**Pull request by wimax-grapl**
This will be merged into the (otherwise hard-to-review, because it's mostly just a revert) https://github.com/grapl-security/grapl/pull/1195

which will be merged into Nomad

which will be merged into Main

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,10 +7,10 @@
 -include .env
 TAG ?= dev
 RUST_BUILD ?= debug
-USERNAME = $(shell id -u -n)
 UID = $(shell id -u)
 GID = $(shell id -g)
 PWD = $(shell pwd)
+GRAPL_ROOT = ${PWD}
```
Comment (wimax-grapl) [inline Makefile:13]: The `PWD` variable actually gets overwritten in some contexts, hence:

File: nomad/local/integration-tests.nomad [hunk 2]
```diff
@@ -170,78 +175,58 @@ job "integration-tests" {
...
-#        read_only   = false
-#      }
-
-
+      user   = var.docker_user
 
       config {
         image = "${var.container_registry}grapl/python-integration-tests:dev"
+        # Pants caches requirements per-user. So when we run a Docker container
+        # with the host's userns, this lets us reuse the pants cache.
```
Comment (jgrillo-grapl) [inline nomad/local/integration-tests.nomad:185]: fascinating!

File: src/python/Dockerfile [hunk 3]
```diff
@@ -66,3 +66,6 @@ RUN mkdir -p /tmp/pants/setup && chmod -R 777 /tmp/pants/setup
 ENV PANTS_NAMED_CACHES_DIR=/tmp/pants/named_caches
 ENV PANTS_LOCAL_STORE_DIR=/tmp/pants/lmdb_store
 ENV PANTS_SETUP_CACHE=/tmp/pants/setup
+
+# grapl-root is mounted by Nomad.
+ENTRYPOINT ["/bin/bash", "-o", "errexit", "-o", "nounset", "-c", "cd /mnt/grapl-root && ./pants filter --filter-target-type='python_tests' :: | xargs ./pants --tag='-needs_work' test --pytest-args='-m integration_test'"]
```
Comment (jgrillo-grapl) [inline src/python/Dockerfile:71]: 👍

**Review by jgrillo-grapl — APPROVED** [review]
LGTM

**Pull request merged**